os/provisioning.md

```diff
@@ -0,0 +1,127 @@
+
+## Config Transpiler
+
+The Container Linux Config Transpiler is the tool that makes it possible for the end user to remain distanced from the details of configuring Container Linux. It's responsible for transforming a Container Linux Config written by an end user into an Ignition Config to be consumed by instances of Container Linux.
+
+To go back to the previous example of configuring etcd, the following config will configure an etcd cluster using the machine's public and private IP addresses:
+
+```container-linux-config:none
+etcd:
+  advertise_client_urls:       http://{PUBLIC_IPV4}:2379
+  initial_advertise_peer_urls: http://{PRIVATE_IPV4}:2380
+  listen_client_urls:          http://0.0.0.0:2379
+  listen_peer_urls:            http://{PRIVATE_IPV4}:2380
+  discovery:                   https://discovery.etcd.io/<token>
+```
+
+It was alluded to earlier that ct requires information about the target environment before it can transform configs which use templating. If this config is passed to ct without any other arguments, ct fails:
+
+```
+$ bin/ct < example.yml
+error: platform must be specified to use templating
+```
+
+This says that the config takes advantage of templating (e.g. `PUBLIC_IPV4`) but ct was invoked without being told what environment to target.
+
+CT can be invoked again and given Amazon EC2 as an example:
+
+```
+$ bin/ct --platform=ec2 < example.yml
+{"ignition":{"version":"2.0.0","config"...
+```
+
+This time, ct successfully runs and produces the following Ignition Config:
+
+```ignition-config
+{
+  "ignition": { "version": "2.0.0" },
+  "systemd": {
+    "units": [{
+      "name": "etcd-member.service",
+      "enable": true,
+      "dropins": [{
+        "name": "20-clct-etcd-member.conf",
+        "contents": "[Unit]\nRequires=coreos-metadata.service\nAfter=coreos-metadata.service\n\n[Service]\nEnvironmentFile=/run/metadata/coreos\nExecStart=\nExecStart=/usr/lib/coreos/etcd-wrapper $ETCD_OPTS \\\n  --listen-peer-urls=\"http://${COREOS_EC2_IPV4_LOCAL}:2380\" \\\n  --listen-client-urls=\"http://0.0.0.0:2379\" \\\n  --initial-advertise-peer-urls=\"http://${COREOS_EC2_IPV4_LOCAL}:2380\" \\\n  --advertise-client-urls=\"http://${COREOS_EC2_IPV4_PUBLIC}:2379\" \\\n  --discovery=\"https://discovery.etcd.io/\u003ctoken\u003e\""
+      }]
+    }]
+  }
+}
+```
+
+This Ignition Config enables and configures etcd as specified in the above Container Linux Config. This can be more easily seen if the contents of the etcd drop-in are formatted nicely:
+
+```ini
+[Unit]
+Requires=coreos-metadata.service
+After=coreos-metadata.service
+
+[Service]
+EnvironmentFile=/run/metadata/coreos
+ExecStart=
```

Comment:
Yes. `ExecStart` is a list in systemd, so every assignment appends something to that list. A non-zero list is only valid for service types of `oneshot`, so we clear the list by assigning null.